src/standalone/Vegetation/Canopy.jl

```diff
@@ -411,15 +425,33 @@ function ClimaLSM.make_update_aux(
         q::FT = canopy.atmos.q(t)
 
         # update radiative transfer
+        RT = canopy.radiative_transfer
         K = extinction_coeff(ld, θs)
-        PAR .= compute_PAR(canopy.radiative_transfer, canopy.radiation, t)
-        @. APAR = plant_absorbed_ppfd(
-            canopy.radiative_transfer,
-            PAR / (energy_per_photon * N_a),
+        RTP = RT.parameters
+        PAR .= compute_PAR(RT, canopy.radiation, t)
+        NIR .= compute_NIR(RT, canopy.radiation, t)
+        # τ is defined only for TwoStream model, not for BeerLambert
+        τ_PAR = typeof(RT) == TwoStreamModel ? RTP.τ_PAR_leaf : FT(0)
+        τ_NIR = typeof(RT) == TwoStreamModel ? RTP.τ_NIR_leaf : FT(0)
+        @. APAR = plant_absorbed_pfd(
```

Comment:
Im wondering if it makes more sense to pass RT, PAR/(energy per photon * Na), LAI, K, theta_s, ground_albedo only as args, and then inside the `plant_absorbed_pfd` function, we can unpack what we need from RT.parameters.

then, because we have two methods for this function, we can unpack tau in one case and not the other.

Reply:
oh, hm, nvm. this wont work because you need to know which \tau to unpack, etc. (NIR or PAR).

Reply:
Im not sure. perhaps we can leave it alone for now because we'll revisit this in the future at some point if we have an arbitrary number of wavelength bands

Reply:
but if you have ideas about how to streamline, please feel free!

Reply:
The only problem is how to tell `plant_absorbed_pfd` which reflectivity and transmittance to use. When we call it for PAR, we use the PAR values, for NIR, the NIR values. We would need to pass another argument which specified which stream to calculate for.

I originally tried to solve this by writing `calculate_absorbances` which dispatched off the type of RT and made the calls to `plant_absorbed_pfd` with the appropriate arguments and then returned a tuple of (APAR, ANIR), but I couldn't get this to work neatly with the broadcasting.

Reply:
nice idea! i think we have an example of something similar here: https://github.com/CliMA/ClimaLSM.jl/blob/386572b94019657ee24b6463ee81c79e3dfa3868/src/standalone/Soil/boundary_conditions.jl#L196
We do the broadcasting inside of this function, and return the fields. 

when we use the function, we dont broadcast: https://github.com/CliMA/ClimaLSM.jl/blob/386572b94019657ee24b6463ee81c79e3dfa3868/src/standalone/Soil/energy_hydrology.jl#L201C13-L201C13

Reply:
Ah! Thanks this helps a lot.